hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -352,6 +352,16 @@ private Object createProxyIfNecessary(
 			}
 			return existing;
 		}
+
```

Comment:
IMO it's a little surprising to be loading from the 2nd-level cache here. The need to do this is really an implementation detail that is only important when the entity class has subclasses, (i.e. `persister.getClassMetadata().hasSubclasses() == true`).

I think it's worthwhile to add a comment explaining why it's being done here.

I think it's also worthwhile to only consult the 2L cache in when the entity class has subclasses, to avoid the overhead in other cases.